akka-stream/src/main/scala/akka/stream/FlowMonitor.scala

```diff
@@ -0,0 +1,37 @@
+package akka.stream
+
+import akka.stream.FlowMonitor.StreamState
+
+/**
+ * Used to monitor the state of a stream
+ * @tparam T Type of messages passed by the stream
+ */
+trait FlowMonitor[+T] {
+  def state: StreamState[T]
+}
+
+object FlowMonitor {
+  sealed trait StreamState[+U]
+
+  /**
+   * Stream was created, but no events have passed through it
+   */
+  case object Initialized extends StreamState[Nothing]
```

Comment:
we need a java api for this one, I'm not sure what naming convention we have in streams, but some kind of method is needed in `object FlowMonitor`:

```
/**
 * JAVA API
*/
def getInitializedInstance = Initialized
```